[dynamo] clear out traced frames at the start of `test_log_traced_frames`

StrongerXi opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #145640

The test was being flaky in CI, and this patch fixes it.

Fixes #137461.

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @chenyang78 @kadeng @chauhang @amjames

StrongerXi committed:
Update

[ghstack-poisoned]

williamwen42 reviewed (commented):
I also cleared the dynamo_tls in this test that I added: https://github.com/pytorch/pytorch/pull/144885/files

Can you also update that callsite?

StrongerXi commented:
> I also cleared the dynamo_tls in this test that I added: https://github.com/pytorch/pytorch/pull/144885/files
> 
> Can you also update that callsite?

Nice, updated.

StrongerXi committed:
Update

[ghstack-poisoned]

StrongerXi commented:
@pytorchbot merge
